src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/BooleanFieldIndexProvider.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentFields.Fields;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using OrchardCore.Data;
+using YesSql.Indexes;
+
+namespace OrchardCore.ContentFields.Indexing
+{
+    // Remark: 
+
+    public class BooleanFieldIndex : MapIndex
+    {
+        public string FieldName { get; set; }
```

Comment:
Like in Taxonomies you need the ContentType and the ContentPart properties on all indices